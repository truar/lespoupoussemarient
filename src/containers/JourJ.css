/* Extra small devices (phones, less than 768px) */
/* No media query since this is the default in Bootstrap */

.JourJ {
    text-shadow: none;
    color: black;
    position: relative;
}

.JourJ .transparent-background {
    display: none;
}

.JourJ .carousel-caption {
    background-color: rgba(0,0,0,0.5);
    border-radius: 2px;
    padding: 20px;
    bottom: auto!important;
}

.JourJ .PageTitle {
    display: none;
    text-shadow: 1px 2px 2px black;
}

.JourJ .carousel-item img {
    width: 100%;
}

.JourJ .carousel-control-next, .JourJ .carousel-control-prev {
    cursor: pointer;
    width: 10%;
}

.JourJ .carousel-control-next:focus, .JourJ .carousel-control-next:hover, .JourJ .carousel-control-prev:focus, .JourJ .carousel-control-prev:hover {
    opacity: .5;
}

.JourJ .carousel-control-prev {
    background-image: linear-gradient(to left, rgba(0,0,0,0), rgba(0,0,0,1));
}

.JourJ .carousel-control-next {
    background-image: linear-gradient(to right, rgba(0,0,0,0), rgba(0,0,0,1));
}

.JourJ .carousel-item {
    height: 50vh;
}

.JourJ .Union {
    background-image: url('../images/union.jpg');
}

.JourJ .Ceremonie {
    background-image: url('../images/ceremonie.jpg');
}

.JourJ .VinHonneur {
    background-image: url('../images/vin-honneur.jpg');
}

.JourJ .Sport {
    background-image: url('../images/sport.jpg');
}

.JourJ .Diner {
    background-image: url('../images/diner.jpg');
}

.JourJ .Soiree {
    background-image: url('../images/soiree.jpg');
}
.JourJ .Brunch {
    background-image: url('../images/brunch.jpg');
}

.JourJ .carousel .carousel-indicators, .JourJ .carousel .carousel-control-prev, .JourJ .carousel .carousel-control-next {
    display: none;
}

.JourJ .carousel-caption img {
    display: none;
}

.JourJ .carousel-caption h3 {
    font-size: 25px;
}

.JourJ .carousel-caption p {
    font-size: 14px;
}

.JourJ .d-none {
    display: block!important;
}

.JourJ .upAndDown {
    display: none;
}

@keyframes moveUpAndDown {
    0% {margin-top: -50px;}
    50% {margin-top: -30px;}
    100% {margin-top: -50px;}
}

@media (min-width: 768px) {
    .carousel .carousel-item:first-child h3 {
        background : url('../images/union-profile.png') no-repeat top;
        background-size: 25%;
        padding-top: 27%;
    }

    .JourJ .carousel-caption h3 {
        font-size: 40px;
    }

    .JourJ .carousel-caption p {
        font-size: 25px;
    }
}

@media (min-width: 992px) {
    .JourJ div.upAndDown { 
        display: block;
        color: white;
        animation: moveUpAndDown 3s infinite;
        position: absolute;
        width: 9%;
        margin-top: -20px;
        right: 0%;
        font-size: 1.2rem;
        top: 48%;
    }

    .JourJ .PageTitle {
        margin-left: 45px;
        margin-top: 1rem;
        position: absolute;
        z-index: 100;
        text-align: center;
        width: 100%;
        display: block;
    }

    .JourJ .carousel-indicators li:nth-child(1) {
        margin-left: 90px;
    }

    .JourJ .carousel-item {
        height: 100vh;
    }

    .JourJ .carousel .carousel-indicators, .JourJ .carousel .carousel-control-prev, .JourJ .carousel .carousel-control-next {
        display: flex;
    }

    .JourJ .carousel .carousel-control-prev {
        margin-left: 90px;
    }

    .JourJ .carousel-caption {
        position: relative;
        right: 0;
        left: 0;
        bottom: 0;
        width: 70%;
        margin-left: 90px;
    }

    .JourJ .carousel-item-next, .JourJ .carousel-item-prev, .JourJ .carousel-item.active {
        display: flex;
        justify-content: center;
    }
}

@media (max-width: 992px) {
    .JourJ .carousel .carousel-item{
        display: flex; 
        flex-direction: column;
        justify-content: center;
        background-size: cover;
        background-position: center;
        bottom: auto!important;
    }

}
