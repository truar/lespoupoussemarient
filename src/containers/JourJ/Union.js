import React, { Component } from 'react';
import UnionPhoto from '../../images/union-profile.png';
import SliderElement from '../../components/SliderElement';
import {Carousel} from 'react-bootstrap';
import './Union.css';
import UnionPicture from '../../images/union.jpg'

function Union(props) {
    return (
        <Carousel.Item>               
            <img src={UnionPicture} />
            <Carousel.Caption>
            <h3>First slide label</h3>
            <p>Nulla vitae elit libero, a pharetra augue mollis interdum.</p>
            </Carousel.Caption>
        </Carousel.Item>
    );
}

export default Union;