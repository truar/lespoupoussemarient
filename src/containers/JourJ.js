import React, { Component } from 'react';
import {
    Carousel,
    CarouselItem,
    CarouselControl,
    CarouselIndicators,
  } from 'reactstrap';
import Section from '../components/Section';
import PageTitle from '../components/PageTitle';
import Element from '../components/Element';
import './JourJ.css';

const items = [
    {
        key: 1,
        className: 'Union',
        header: 'Anouk & Thibault',
        text: 'S’unissent pour le meilleur et pour le pire le 31 Août 2019 au domaine de Saint-Jean de Chépy',
        category: 1
    }, {
        key: 2,
        className: 'Ceremonie',
        header: 'Cérémonie',
        text: 'Nous vous donnons rendez-vous à 10h45 au domaine de Saint-Jean de Chépy. La cérémonie débutera à 11h00 sous le chapiteau',
        category: 1
    }, {
        key: 3,
        className: 'VinHonneur',
        header: 'Le vin d\'honneur',
        text: 'Se déroulera de 13h00 à 15h00. Une activité « guest book » aura lieu dans l’après-midi. Nous aimerions que chaque invité, couple, famille apporte une photo qui lui est chère',
        category: 1
    }, {
        key: 4,
        className: 'Sport',
        header: 'Activités sportives',
        text: 'De 15h00 à 18h00. N’oubliez pas de prévoir une tenue adéquate (Vestiaire et douches sur place)',
        category: 1
    }, {
        key: 5,
        className: 'Diner',
        header: 'Diner',
        text: 'Vous pourrez vous installer dans la salle principale à partir de 19h00. Nous vous proposons de déguster un repas 100% végétarien',
        category: 2
    }, {
        key: 6,
        className: 'Soiree',
        header: 'Soirée',
        text: 'C\'est le moment d’enflammer la piste de danse ! Les festivités s’arrêteront vers 4h00 (N\'oubliez pas de consulter la page hébergements pour savoir ou dormir)',
        category: 2
    }, {
        key: 7,
        className: 'Brunch',
        header: 'Brunch',
        text: 'Dimanche 1 Septembre à partir de 11h00. La journée sera placée sous le thème de la détente. N’oubliez pas vos maillots de bain',
        category: 3
    }
];

class JourJ extends Component {
    constructor(props) {
        super(props);
        this.state = { activeIndex: 0 };
        this.keyboard = true;
        this.interval = false;
        this.next = this.next.bind(this);
        this.previous = this.previous.bind(this);
        this.goToIndex = this.goToIndex.bind(this);
        this.onExiting = this.onExiting.bind(this);
        this.onExited = this.onExited.bind(this);
        this.itemsForProfile = items.filter((item) => {
            return item.category <= props.profile;
        });
      }
    
      onExiting() {
        this.animating = true;
      }
    
      onExited() {
        this.animating = false;
      }
    
      next() {
        if (this.animating) return;
        const nextIndex = this.state.activeIndex === this.itemsForProfile.length - 1 ? 0 : this.state.activeIndex + 1;
        this.setState({ activeIndex: nextIndex });
      }
    
      previous() {
        if (this.animating) return;
        const nextIndex = this.state.activeIndex === 0 ? this.itemsForProfile.length - 1 : this.state.activeIndex - 1;
        this.setState({ activeIndex: nextIndex });
      }
    
      goToIndex(newIndex) {
        if (this.animating) return;
        this.setState({ activeIndex: newIndex });
      }

      render() {
        const { activeIndex } = this.state;
        
        const slides = this.itemsForProfile.map((item, i) => {
            return (
                <CarouselItem
                    onExiting={this.onExiting}
                    onExited={this.onExited}
                    key={item.key}
                    className={item.className}
                >
                    <Element {...item}/>
                </CarouselItem>
            );
        });

        return (
            <Section id="JourJ" className="JourJ" {...this.props}>
                <PageTitle title="Le Jour J" />
                <Carousel
                    activeIndex={activeIndex}
                    next={this.next}
                    previous={this.previous}
                    keyboard={this.keyboard}
                    interval={this.interval}
                >
                    <CarouselIndicators items={this.itemsForProfile} activeIndex={activeIndex} onClickHandler={this.goToIndex} />
                    {slides}
                    <CarouselControl direction="prev" directionText="Previous" onClickHandler={this.previous} />
                    <CarouselControl direction="next" directionText="Next" onClickHandler={this.next} />
                </Carousel>
            </Section>
        );
      }
}

export default JourJ;