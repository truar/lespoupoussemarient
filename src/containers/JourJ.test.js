import React from 'react';
import JourJ from './JourJ';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<JourJ />);
  expect(component).toMatchSnapshot();
});
