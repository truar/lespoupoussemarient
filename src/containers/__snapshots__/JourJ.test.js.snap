// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JourJ />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageTitle
          title="Le Jour J"
        />,
        <Carousel
          activeIndex={0}
          interval={false}
          keyboard={true}
          next={[Function]}
          pause="hover"
          previous={[Function]}
          slide={true}
        >
          <CarouselIndicators
            activeIndex={0}
            items={Array []}
            onClickHandler={[Function]}
          />
          <CarouselControl
            direction="prev"
            directionText="Previous"
            onClickHandler={[Function]}
          />
          <CarouselControl
            direction="next"
            directionText="Next"
            onClickHandler={[Function]}
          />
        </Carousel>,
        <div
          className="upAndDown"
        >
          Cliquez ici
        </div>,
      ],
      "className": "JourJ",
      "id": "JourJ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "Le Jour J",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeIndex": 0,
          "children": Array [
            <CarouselIndicators
              activeIndex={0}
              items={Array []}
              onClickHandler={[Function]}
            />,
            Array [],
            <CarouselControl
              direction="prev"
              directionText="Previous"
              onClickHandler={[Function]}
            />,
            <CarouselControl
              direction="next"
              directionText="Next"
              onClickHandler={[Function]}
            />,
          ],
          "interval": false,
          "keyboard": true,
          "next": [Function],
          "pause": "hover",
          "previous": [Function],
          "slide": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeIndex": 0,
              "items": Array [],
              "onClickHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "direction": "prev",
              "directionText": "Previous",
              "onClickHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "direction": "next",
              "directionText": "Next",
              "onClickHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cliquez ici",
          "className": "upAndDown",
        },
        "ref": null,
        "rendered": "Cliquez ici",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageTitle
            title="Le Jour J"
          />,
          <Carousel
            activeIndex={0}
            interval={false}
            keyboard={true}
            next={[Function]}
            pause="hover"
            previous={[Function]}
            slide={true}
          >
            <CarouselIndicators
              activeIndex={0}
              items={Array []}
              onClickHandler={[Function]}
            />
            <CarouselControl
              direction="prev"
              directionText="Previous"
              onClickHandler={[Function]}
            />
            <CarouselControl
              direction="next"
              directionText="Next"
              onClickHandler={[Function]}
            />
          </Carousel>,
          <div
            className="upAndDown"
          >
            Cliquez ici
          </div>,
        ],
        "className": "JourJ",
        "id": "JourJ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "Le Jour J",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeIndex": 0,
            "children": Array [
              <CarouselIndicators
                activeIndex={0}
                items={Array []}
                onClickHandler={[Function]}
              />,
              Array [],
              <CarouselControl
                direction="prev"
                directionText="Previous"
                onClickHandler={[Function]}
              />,
              <CarouselControl
                direction="next"
                directionText="Next"
                onClickHandler={[Function]}
              />,
            ],
            "interval": false,
            "keyboard": true,
            "next": [Function],
            "pause": "hover",
            "previous": [Function],
            "slide": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeIndex": 0,
                "items": Array [],
                "onClickHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "direction": "prev",
                "directionText": "Previous",
                "onClickHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "direction": "next",
                "directionText": "Next",
                "onClickHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cliquez ici",
            "className": "upAndDown",
          },
          "ref": null,
          "rendered": "Cliquez ici",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
